### ISSUE TITLE
ValueError: invalid mode: 'rU'

### ISSUE THREAD

**Issue by alexzaccaron** [davidemms/OrthoFinder#760]
Hi,

I am getting the error `ValueError: invalid mode: 'rU'` when trying to run the current OrthoFinder v2.5.4 at Bioconda. The error shown below occurs after OrthoFinder finds the orthogroups, and starts to infer gene and species trees. I have the same version of OrthoFinder v2.5.4 installed in an older conda environment, which finishes successfully using the same data set. Perhaps there is something wrong with the current release at Bioconda?

```
Traceback (most recent call last):
  File "/home/azacca/miniconda3/envs/orthofinder/bin/orthofinder", line 7, in <module>
    main(args)
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 1778, in main
    GetOrthologues(speciesInfoObj, options, prog_caller)
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/__main__.py", line 1540, in GetOrthologues
    orthologues.OrthologuesWorkflow(speciesInfoObj.speciesToUse, 
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/orthologues.py", line 999, in OrthologuesWorkflow
    spTreeFN_ids, qSTAG = db.RunAnalysis(userSpeciesTree == None)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/orthologues.py", line 481, in RunAnalysis
    stag.Run_ForOrthoFinder(files.FileHandler.GetOGsTreeDir(), files.FileHandler.GetWorkingDirectory_Write(), self.ogSet.seqsInfo.speciesToUse, spTreeFN_ids)
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/stag.py", line 262, in Run_ForOrthoFinder
    ProcessTrees(dir_in, dir_matrices, dir_trees_out, gene_to_species, qVerbose=False, qForOF=True)
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/stag.py", line 191, in ProcessTrees
    t = tree.Tree(fn)
        ^^^^^^^^^^^^^
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/tree.py", line 221, in __init__
    read_newick(newick, root_node = self, format=format)
  File "/home/azacca/miniconda3/envs/orthofinder/bin/scripts_of/newick.py", line 208, in read_newick
    nw = open(newick, 'rU').read()
         ^^^^^^^^^^^^^^^^^^
ValueError: invalid mode: 'rU'
```

Here's the code I am running:

```
# install OrthoFinder
conda create -n orthofinder
conda activate orthofinder
conda install -c bioconda orthofinder=2.5.4


# get data from NCBI
mkdir data/
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/009/045/GCF_000009045.1_ASM904v1/GCF_000009045.1_ASM904v1_protein.faa.gz -O data/GCF_000009045.1.fasta.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/004/124/315/GCF_004124315.2_ASM412431v2/GCF_004124315.2_ASM412431v2_protein.faa.gz -O data/GCF_004124315.2.fasta.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/020/519/665/GCF_020519665.1_ASM2051966v1/GCF_020519665.1_ASM2051966v1_protein.faa.gz -O data/GCF_020519665.1.fasta.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/012/955/605/GCF_012955605.1_ASM1295560v1/GCF_012955605.1_ASM1295560v1_protein.faa.gz -O data/GCF_012955605.1.fasta.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/220/285/GCF_002220285.1_ASM222028v1/GCF_002220285.1_ASM222028v1_protein.faa.gz -O data/GCF_002220285.1.fasta.gz

# decompress
gunzip data/*.gz

# run OrthoFinder
orthofinder -t 8 -f data/
```

**Comment by austintdaigle** [comment]
@alexzaccaron I had the same error using the latest conda install of orthofinder. I was able to solve this problem in my install by finding the offending line of code in newick.py ('anaconda3/envs/orthofinder/bin/scripts_of/newick.py') on line 208. When I changed 
`nw = open(newick, 'rU').read()`
to 
`nw = open(newick, 'r').read()`
the program began to work as expected. I think this is because the -U option in open() has been depreciated for a while in python, and is now the default behavior.
